spec/rubocop/cop/rspec/predicate_spec.rb

```diff
@@ -0,0 +1,116 @@
+# encoding: utf-8
+
+describe RuboCop::Cop::RSpec::Predicate, :config do
+  subject(:cop) { described_class.new(config) }
+
+  context 'when EnforcedStyle is boolean' do
+    let(:cop_config) { { 'EnforcedStyle' => 'equality' } }
+
+    TESTS = {
+      'expect(foo.bar).to be_valid' =>
+        'expect(foo.bar.valid?).to eq(true)',
+      'expect({ foo: :bar }).to have_key(:foo)' =>
+        'expect({ foo: :bar }.key?(:foo)).to eq(true)',
+      'expect({ foo: :bar }).to have_attributes(:foo, :baz)' =>
+        'expect({ foo: :bar }.has_attributes?(:foo, :baz)).to eq(true)',
+      'expect([]).to respond_to(:foo)' =>
+        'expect([].respond_to?(:foo)).to eq(true)',
+      'expect([]).not_to be_a(Array)' =>
+        'expect([].is_a?(Array)).to eq(false)',
+      'expect([:foo]).to include(:foo)' =>
+        'expect([:foo].include?(:foo)).to eq(true)'
+    }.freeze
+
+    TESTS.each do |offense, suggestion|
+      it "flags `#{offense}`" do
```

Comment:
This came after you originally opened this PR, but I highly recommend using this matcher that I added which makes tests like these way more clear and readable. This is what these tests would look like:

```ruby

it 'flags violations' do
  expect_violation(<<-RUBY)
    expect(foo.bar).to be_valid
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect(foo.bar.valid?).to eq(true)`
    expect({ foo: :bar }).to have_key(:foo)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect({ foo: :bar }.key?(:foo)).to eq(true)`
    expect({ foo: :bar }).to have_attributes(:foo, :baz)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect({ foo: :bar }.has_attributes?(:foo, :baz)).to eq(true)`
    expect([]).to respond_to(:foo)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect([].respond_to?(:foo)).to eq(true)`
    expect([]).not_to be_a(Array)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect([].is_a?(Array)).to eq(false)`
    expect([:foo]).to include(:foo)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use predicate matcher `expect([:foo].include?(:foo)).to eq(true)`
  RUBY
end
```